### ISSUE TITLE
Promised to not declare war, but even 2 turns after it ended I still can't declare without being seen as a traitor

### ISSUE THREAD

**Issue by danielsolaro** [LoneGazebo/Community-Patch-DLL#11247]
_1. Mod Version (X.Y.Z). Current Version: 4.15.2 with EUI and normal amount of civs

_3. List of Other Mods_
I made a rar with all mods that were activated. https://www.mediafire.com/file/zmquvuwyrlp4bg3/MODS.rar/file

_4. Describe the Issue_
Sumer* questions my military on his border, I promise to not declare way, turns go by and promise ends, text disappears from his portrait UI but declaring war still considers me a traitor. I reload, give an extra turn and dow, again considered a traitor.
I don't see the text saying I fulfilled the promise despite the extra safety turn after it ended.

* I'm pretty sure I had it happen with another civ before but I was tired and a bit confused, just left it as I didn't push that game further. Now that I can see it happening again I'm 99% sure its not specific to the sumer.

______________________________________________________________________________________
_5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_
Saves included in the RAR file. Autosave224 loads on turn 225 which is the last one I can't dow according to the UI. Turns 226 and 227 are when I try to dow.

_6. Logs (ALWAYS ATTACH THESE IF POSSIBLE)_
Attached.

_8. Steps to reproduce the Issue (Optional)_
Try to dow after the message about promising to not declare war disappears (turns 226+)

**Comment by coinich** [comment]
I saw this once myself, though I have no saves or artifacts to help explain or back it up.  What speed are you playing on?  Is this a speed modifier not matching the text string?

**Comment by LStuber** [comment]
I have the same issue, with different leader and all default options (standard game speed in particular).

In that game, Ramesses caught me trying to sneak attack, but I wasn't ready, so I had to make the promise to not declare war. But we're now more than 100+ turns after I made the promise, and I still haven't seen the message that it's ended, and when I declare war, I'm getting the backstabber penalty from all the other leaders.

Something I suspect, is that if after 50 turns you still have lots of troops at his border, then the AI doesn't trust you, even if the message disappears. Maybe it wants to ask you again to back up your troops for another 50 turns, but the mesage does not appear for some reason? Would that make sense?
